src/aura/CRLP_Rollup/CRLP_RollupController.js

```diff
@@ -0,0 +1,179 @@
+({
+
+    getRollup: function(cmp, event, helper) {
+        console.log("IN GETROLLUP");
+        /**called when the activeRollupId changes, specifically in the rollupRow
+         * queries active rollup ID to populate the full active rollup detail
+         * switches the display to the detail view and sets the width for the buttons
+         * called after user returns to grid since activeRollupId is cleared, null check is necessary**/
+        var activeRollupId = cmp.get("v.activeRollupId");
+
+        if (activeRollupId != null) {
+            var action = cmp.get("c.getRollupById");
+            action.setParams({id: activeRollupId});
+
+            action.setCallback(this, function (response) {
+                var state = response.getState();
+                if (state === "SUCCESS") {
+                    //note: the parsing is important to avoid a shared reference
+                    //todo: try to stringify on server side; only parse on client
+                    //review: https://stackoverflow.com/questions/6605640/javascript-by-reference-vs-by-value
+                    cmp.set("v.activeRollup", JSON.parse(JSON.stringify(response.getReturnValue())));
```

Comment:
We actually struggled with this for a bit, and it wasn't possible to assign them to the same value because of some strange JS/lightning reasons. We needed a cached value to stay the same to avoid another server trip, but if we assigned to the same var then they ended up working as though they were pointing to the same place in memory and both attributes were changing! If we end up using this a lot, I agree that a helper method would greatly improve readability.